src/store-api/src/region_engine.rs

```diff
@@ -44,11 +43,11 @@ pub trait RegionEngine {
     async fn handle_query(
         &self,
         region_id: RegionId,
-        request: QueryRequest,
+        request: ScanRequest,
     ) -> Result<SendableRecordBatchStream, BoxedError>;
 
     /// Retrieve region's metadata.
-    async fn get_metadata(&self, region_id: RegionId) -> Result<RegionMetadata, BoxedError>;
+    async fn get_metadata(&self, region_id: RegionId) -> Result<RegionMetadataRef, BoxedError>;
```

Comment:
What if the region doesn't exist? Will this method return an error?

Reply:
Yes, the implement can return an error if it doesn't exist